core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -177,35 +177,55 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   // Until we have a good workaround for that bug the constructors remain broken out.
 
   /**
-   * Alternative constructor that allows setting common Spark properties directly
-   *
-   * @param master Cluster URL to connect to (e.g. mesos://host:port, spark://host:port, local[4]).
-   * @param appName A name for your application, to display on the cluster web UI.
-   */
-  private[spark] def this(master: String, appName: String) =
-    this(master, appName, null, Nil, Map())
+    * Alternative constructor that allows setting common Spark properties directly
```

Comment:
This looks like a lot of change, but it's really adding one more constructor, reordering them so they can cascade, and then having them call each other, to avoid calls to the deprecated constructor. If the deprecated one goes away I think this is largely unnecessary.